multitask_exp.py

```diff
@@ -0,0 +1,90 @@
+import os
+import tensorflow as tf
+from lib import multitask as mt
+from utils import args_util, training
+
+def main():
+  """Performs a finetuning or domain adaptation experiment."""
+  parser = mt.get_custom_parser()
+  args = parser.parse_args()
+  args.save_path = os.path.abspath(args.save_path)
+  exp_path = os.path.join(args.save_path, args.name)
+  if not os.path.exists(exp_path):
+    os.makedirs(exp_path)
+  args_util.write_config(args, exp_path)
+
+  if len(args.ckpt_path) == 0:
+    args.ckpt_path = None
+  # pretrain the model
+  new_shape = (args.reshape_to, args.reshape_to, 3)
+  model, train_dataset, test_dataset = mt.pretrain(
+      new_shape, data_dir=args.data_dir, save_path=exp_path,
```

Comment:
This many arguments can be a bit problematic to work with. I would suggest using `dataclasses` (possibly a hierarchical structure) to wrap different configuration parameters and possibly pass them between functions that use them.